solutions/cses-1139.mdx

```diff
@@ -0,0 +1,117 @@
+---
+id: cses-1139
+source: CSES
+title: Distinct Colors
+author: Timothy Gao
+---
+
+## Solution 1 - Small to Large Merging
+
+See [here](/plat/merging#merging-data-structures).
+
+## Solution 2 - PURS
+
+Let us consider the Euler Tour of the tree. We flatten the tree into an array, where each node corresponds to a range in this array. Now, we are essentially tasked to find the number of distinct values in a range $N$ times (once for each node).
+
+Let us consider iterating through the Euler Tour array from left to right. 
+When we consider a node at index $i$ in the Euler Tour array, its subtree range will be $[j, i]$, where $j\leq i$.
+Now, let's focus on a single color. Notice that if there are multiple colors $c$ to the left of a certain index $i$,
+only $c$'s most rightmost occurrence is relevant. This is because any non-rightmost occurrence of $c$
```

Comment:
capitalize "only"

Reply:
~~did grammarly not catch this?~~

Reply:
maybe you want to mention that "we only want to count $c$ once so we can choose to only count the most rightmost occurence"

Reply:
"only" is not the start of a new sentence 👀

Reply:
oops